rbi/core/kernel.rbi

```diff
@@ -1874,21 +1874,21 @@ module Kernel
   sig {returns(Float)}
   sig do
     params(
-        arg0: Integer,
+        arg0: T.any(Integer, T::Range[Integer]),
     )
     .returns(Integer)
   end
   sig do
     params(
-        arg0: T::Range[Integer],
+        arg0: T.any(T.nilable(Float), T::Range[Float]),
```

Comment:
Changed this to a nilable float because `Kernel.rand(nil)` is also valid but [currently misbehaving](https://sorbet.run/#%23%20typed%3A%20true%0AKernel.rand(nil)).

```
irb(main):001:0> Kernel.rand(nil)
=> 0.7324702695024953
```

See [here](https://ruby-doc.org/core-3.0.0/Kernel.html#method-i-rand).

Interestingly, this isn't the case for `Random`'s rand methods.